### PR TITLE
added product details page

### PR DESCRIPTION
**Pull request by Ksenaaa**

### CODE REVIEW

File: src/components/productDetails/ProductDetails.tsx [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+import { CustomButton } from 'components/customButton/CustomButton';
+import { Product } from 'model/Product';
+import { RootStackNavigatorParamsList } from 'model/rootStackNavigatorParamsList';
+import useMaterialStore from 'store/MaterialStore';
+import { productColors } from 'utils/helpers/productColors';
+import { useToggle } from 'utils/hooks/toggleHook';
+
+import { stylesProductDetails as styles } from './ProductDetails.styles';
+import { CategoryDetails } from './categoryDetails/CategoryDetails';
+import { ProductImgs } from './productImgs/ProductImgs';
```
Comment (vell28) [inline src/components/productDetails/ProductDetails.tsx:14]: import ordering

File: src/utils/hooks/toggleHook.ts [hunk 2]
```diff
@@ -0,0 +1,13 @@
+import { useCallback, useState } from 'react';
```
Comment (vell28) [inline src/utils/hooks/toggleHook.ts]: rename file to with `use` at the beginning

File: src/components/productDetails/ProductDetails.tsx (hunk 1 above)
Comment (Ksenaaa) [reply src/components/productDetails/ProductDetails.tsx:14]: fixed

File: src/utils/hooks/toggleHook.ts (hunk 2 above)
Comment (Ksenaaa) [reply src/utils/hooks/toggleHook.ts]: fixed

**Pull request merged**